src/telemetry/telemetryEvents.ts

```diff
@@ -0,0 +1,666 @@
+import type { ExtensionCommand } from '../commands';
+import type { DocumentSourceDetails } from '../documentSource';
+import { DocumentSource } from '../documentSource';
+import type {
+  ExportToPlaygroundError,
+  ParticipantErrorTypes,
+} from '../participant/participantErrorTypes';
+import type {
+  ParticipantCommandType,
+  ParticipantRequestType,
+  ParticipantResponseType,
+} from '../participant/participantTypes';
+import type { ShellEvaluateResult } from '../types/playgroundType';
+import type { NewConnectionTelemetryEventProperties } from './connectionTelemetry';
+import * as vscode from 'vscode';
+
+type PlaygroundFileType = 'other' | 'mongodbjs' | 'mongodb';
+
+type TelemetryFeedbackKind = 'positive' | 'negative' | undefined;
+
+export type InternalPromptPurpose = 'intent' | 'namespace' | undefined;
+
+export type ParticipantTelemetryMetadata = {
+  // The source of the participant prompt - e.g. 'codelens', 'treeview', etc.
```

Comment:
```suggestion
  /** The source of the participant prompt - e.g. 'codelens', 'treeview', etc. */
```
Not super important but using `/**` will make the IDE show these comments on hover